tests/layer1/test_layer1_speed_Novus_100G.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+from abstract_open_traffic_generator.port import *
+from abstract_open_traffic_generator.config import *
+from abstract_open_traffic_generator.layer1 import *
+from abstract_open_traffic_generator.control import *
+
+
+@pytest.mark.e2e
+def test_layer1(api, options, utils):
+    """Test that layer1 configuration settings are being applied correctly
+    A user should be able to configure ports with/without locations.
+    The expectation should be if a location is configured the user wants to
+    connect but debug should allow for config creation without location.
+    Ports with no location should not generate an error message.
+    Ports with location should generate an error message if unable to connect.
+
+    Validation: Validate the layer1 properties applied using Restpy
+    """
+    speed_type = {'speed_10_gbps': '10000',
```

Comment:
I know speed 50G is not included in models, and hence would suggest parametrizing the test with supported speeds (so we can opt out of any given speed).

Reply:
The dictionary is not testing for each speed. The dict is used for validation. 

ports and speed will be taken from settings file
    "ports": [
        "10.36.87.215;1;3",
        "10.36.87.215;1;4"
    ],
    "speed": "speed_100_gbps",

The port location and speed should be user's input as 40G&100G takes same port location